File: chatuikit/src/main/java/com/cometchat/chatuikit/shared/views/aiconversationsummary/view/AIConversationSummaryAdapter.java
package com.cometchat.chatuikit.shared.views.aiconversationsummary.view;

import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import androidx.annotation.ColorInt;
import androidx.annotation.Dimension;
import androidx.annotation.NonNull;
import androidx.annotation.StyleRes;
import androidx.recyclerview.widget.RecyclerView;

import com.cometchat.chatuikit.R;
import com.cometchat.chatuikit.databinding.CometchatAiConversationSummaryRowBinding;

import java.util.ArrayList;
import java.util.List;

public class AIConversationSummaryAdapter extends RecyclerView.Adapter<AIConversationSummaryAdapter.ViewHolder> {
    private static final String TAG = AIConversationSummaryAdapter.class.getSimpleName();
    private List<String> list;
    private Drawable itemBackgroundDrawable;
    private @ColorInt int itemBackgroundColor;
    private @ColorInt int itemStrokeColor;
    private @Dimension int itemStrokeWidth;
    private @Dimension int itemCornerRadius;
    private @ColorInt int itemTextColor;
    private @StyleRes int itemTextAppearance;

    public AIConversationSummaryAdapter() {
        list = new ArrayList<>();
    }

    /**
     * Called when RecyclerView needs a new {@link ViewHolder} of the given type to
     * represent an item.
     *
     * <p>
     * This new ViewHolder should be constructed with a new View that can represent
     * the items of the given type. You can either create a new View manually or
     * inflate it from an XML layout file.
     *
     * <p>
     * The new ViewHolder will be used to display items of the adapter using
     * {@link #onBindViewHolder(ViewHolder, int, List)}. Since it will be re-used to
     * display different items in the data set, it is a good idea to cache
     * references to sub views of the View to avoid unnecessary
     * {@link View#findViewById(int)} calls.
     *
     * @param parent   The ViewGroup into which the new View will be added after it is
     *                 bound to an adapter position.
     * @param viewType The view type of the new View.
     * @return A new ViewHolder that holds a View of the given view type.
     * @see #getItemViewType(int)
     * @see #onBindViewHolder(ViewHolder, int)
     */
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        CometchatAiConversationSummaryRowBinding binding = CometchatAiConversationSummaryRowBinding.inflate(LayoutInflater.from(parent.getContext()),
                                                                                                            parent,
                                                                                                            false
        );
        return new ViewHolder(binding.getRoot());
    }

    /**
     * Called by RecyclerView to display the data at the specified position. This
     * method should update the contents of the {@link ViewHolder#itemView} to
     * reflect the item at the given position.
     *
     * <p>
     * Note that unlike {@link ListView}, RecyclerView will not call this method
     * again if the position of the item changes in the data set unless the item
     * itself is invalidated or the new position cannot be determined. For this
     * reason, you should only use the <code>position</code> parameter while
     * acquiring the related data item inside this method and should not keep a copy
     * of it. If you need the position of an item later on (e.g. in a click
     * listener), use {@link ViewHolder#getBindingAdapterPosition()} which will have
     * the updated adapter position.
     *
     * <p>
     * Override {@link #onBindViewHolder(ViewHolder, int, List)} instead if Adapter
     * can handle efficient partial bind.
     *
     * @param holder   The ViewHolder which should be updated to represent the contents
     *                 of the item at the given position in the data set.
     * @param position The position of the item within the adapter's data set.
     */
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bindView(list.get(position));
    }

    /**
     * Returns the total number of items in the data set held by the adapter.
     *
     * @return The total number of items in this adapter.
     */
    @Override
    public int getItemCount() {
        return list.size();
    }

    public void setList(List<String> list) {
        this.list = list;
        notifyDataSetChanged();
    }

    public void setItemBackgroundColor(@ColorInt int itemBackgroundColor) {
        this.itemBackgroundColor = itemBackgroundColor;
    }

    public void setItemStrokeColor(@ColorInt int itemStrokeColor) {
        this.itemStrokeColor = itemStrokeColor;
    }

    public void setItemStrokeWidth(@Dimension int itemStrokeWidth) {
        this.itemStrokeWidth = itemStrokeWidth;
    }

    public void setItemCornerRadius(@Dimension int itemCornerRadius) {
        this.itemCornerRadius = itemCornerRadius;
    }

    public void setItemTextColor(@ColorInt int itemTextColor) {
        this.itemTextColor = itemTextColor;
    }

    public void setItemTextAppearance(@StyleRes int itemTextAppearance) {
        this.itemTextAppearance = itemTextAppearance;
    }

    public void setItemBackgroundDrawable(Drawable itemBackgroundDrawable) {
        this.itemBackgroundDrawable = itemBackgroundDrawable;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final CometchatAiConversationSummaryRowBinding binding;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            binding = CometchatAiConversationSummaryRowBinding.bind(itemView);
        }

        public void bindView(String reply) {
            binding.tvMessage.setText(reply);
            binding.messageCard.setCardBackgroundColor(itemBackgroundColor);
            binding.messageCard.setStrokeColor(itemStrokeColor);
            binding.messageCard.setStrokeWidth(itemStrokeWidth);
            binding.messageCard.setRadius(itemCornerRadius);
            binding.tvMessage.setTextColor(itemTextColor);
            binding.tvMessage.setTextAppearance(itemTextAppearance);
            if (itemBackgroundDrawable != null) binding.messageCard.setBackgroundDrawable(itemBackgroundDrawable);
            itemView.setTag(R.string.cometchat_reply_lowercase, reply);
        }
    }
}

File: chatuikit-kotlin/src/main/java/com/cometchat/uikit/kotlin/presentation/shared/aiconversationsummary/AIConversationSummaryAdapter.kt
package com.cometchat.uikit.kotlin.presentation.shared.aiconversationsummary

import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.annotation.ColorInt
import androidx.annotation.Dimension
import androidx.annotation.StyleRes
import androidx.recyclerview.widget.RecyclerView
import com.cometchat.uikit.kotlin.R
import com.cometchat.uikit.kotlin.databinding.CometchatAiConversationSummaryRowBinding

/**
 * AIConversationSummaryAdapter is a RecyclerView adapter for displaying
 * AI-generated conversation summary text.
 *
 * Unlike the ConversationStarterAdapter which displays multiple clickable options,
 * this adapter typically displays a single summary text item that provides
 * an overview of the conversation history.
 *
 * The adapter supports customization of item appearance including background color,
 * corner radius, stroke, and text styling.
 */
class AIConversationSummaryAdapter : RecyclerView.Adapter<AIConversationSummaryAdapter.ViewHolder>() {

    companion object {
        private val TAG = AIConversationSummaryAdapter::class.java.simpleName
    }

    private val list: MutableList<String> = mutableListOf()

    // Item style properties
    private var itemBackgroundDrawable: Drawable? = null
    @ColorInt private var itemBackgroundColor: Int = 0
    @ColorInt private var itemStrokeColor: Int = 0
    @Dimension private var itemStrokeWidth: Int = 0
    @Dimension private var itemCornerRadius: Int = 0
    @ColorInt private var itemTextColor: Int = 0
    @StyleRes private var itemTextAppearance: Int = 0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = CometchatAiConversationSummaryRowBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindView(list[position])
    }

    override fun getItemCount(): Int = list.size

    /**
     * Updates the list of summary items.
     *
     * @param newList The new list of summary strings (typically a single item).
     */
    fun setList(newList: List<String>) {
        list.clear()
        list.addAll(newList)
        notifyDataSetChanged()
    }

    /**
     * Sets the background color for each item.
     */
    fun setItemBackgroundColor(@ColorInt itemBackgroundColor: Int) {
        this.itemBackgroundColor = itemBackgroundColor
    }

    /**
     * Sets the stroke color for each item.
     */
    fun setItemStrokeColor(@ColorInt itemStrokeColor: Int) {
        this.itemStrokeColor = itemStrokeColor
    }

    /**
     * Sets the stroke width for each item.
     */
    fun setItemStrokeWidth(@Dimension itemStrokeWidth: Int) {
        this.itemStrokeWidth = itemStrokeWidth
    }

    /**
     * Sets the corner radius for each item.
     */
    fun setItemCornerRadius(@Dimension itemCornerRadius: Int) {
        this.itemCornerRadius = itemCornerRadius
    }

    /**
     * Sets the text color for each item.
     */
    fun setItemTextColor(@ColorInt itemTextColor: Int) {
        this.itemTextColor = itemTextColor
    }

    /**
     * Sets the text appearance for each item.
     */
    fun setItemTextAppearance(@StyleRes itemTextAppearance: Int) {
        this.itemTextAppearance = itemTextAppearance
    }

    /**
     * Sets the background drawable for each item.
     */
    fun setItemBackgroundDrawable(itemBackgroundDrawable: Drawable?) {
        this.itemBackgroundDrawable = itemBackgroundDrawable
    }

    /**
     * ViewHolder for conversation summary items.
     */
    inner class ViewHolder(
        private val binding: CometchatAiConversationSummaryRowBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        /**
         * Binds the summary text to the view and applies styling.
         *
         * @param summary The conversation summary text to display.
         */
        fun bindView(summary: String) {
            binding.tvMessage.text = summary
            binding.messageCard.setCardBackgroundColor(itemBackgroundColor)
            binding.messageCard.strokeColor = itemStrokeColor
            binding.messageCard.strokeWidth = itemStrokeWidth
            binding.messageCard.radius = itemCornerRadius.toFloat()
            binding.tvMessage.setTextColor(itemTextColor)
            if (itemTextAppearance != 0) {
                binding.tvMessage.setTextAppearance(itemTextAppearance)
            }
            itemBackgroundDrawable?.let {
                binding.messageCard.background = it
            }
            itemView.setTag(R.string.cometchat_reply_lowercase, summary)
        }
    }
}
